mountpoint-s3-client/src/s3_crt_client/list_objects.rs

```diff
@@ -143,7 +153,9 @@ impl S3CrtClient {
                 .inner
                 .new_request_template("GET", bucket)
                 .map_err(S3RequestError::construction_failure)?;
-
+            message
+                .add_header(&Header::new("x-amz-optional-object-attributes", "RestoreStatus"))
+                .map_err(S3RequestError::construction_failure)?;
```

Comment:
This should _probably_ be optional, because I'm not sure it's supported everywhere, but let's tackle that separately.

Reply:
I suspect you're referring to `Outposts`, what I've noticed in this regard is that [restoration](https://docs.aws.amazon.com/AmazonS3/latest/API/API_RestoreObject.html) itself is not supported there:
> This action is not supported by Amazon S3 on Outposts.

So at least new tests under `s3_tests` feature will fail if they will ever run against Outputs S3 installation.